components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -1076,17 +1077,41 @@ public static boolean isSkipRecoveryWithChallengeQuestionsForInsufficientAnswers
     public static void createAuditMessage(String action, String target, JSONObject dataObject, String result) {
 
         if (!isLegacyAuditLogsDisabled()) {
-            String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
-            if (StringUtils.isBlank(loggedInUser)) {
-                loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+            if (LoggerUtils.isLogMaskingEnable) {
+                target = LoggerUtils.getMaskedContent(target);
```

Comment:
changing the target here directly may affect the methods that are calling this method